### PR TITLE
Add remote payment gateway recommendations initial docs

### LINKED ISSUE #6908: Remote Payment Recommendations: Add documentation

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #6908 

Adds some initial documentation around the remote payment gateway recommendations.  Note that a couple items may be slightly tweaked pending open PRs, but we can address this going forward in those PRs.

### Detailed test instructions:

1. Check that docs look accurate.

### CODE REVIEW

File: docs/features/remote-payments.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# WooCommerce Remote Payment Gateway Recommendations
+
+This feature allows a remote data source to specify the currently recommended payment gateways.  The feature polls the remote JSON file to retrieve a list of recommended and visible gateways based on rulesets to be shown to the user.
+
+Plugins from this source will then walk through an installer step, followed by a connection step with the minimum required fields for setup defined by the downloaded plugin.
+
+## Enabling Remote Payment Gateway Recommendations
+
+This feature is behind a feature flag.  In order for it to run, the `remote-payment-methods` must be set to `true` in `~/config/{environment}.json`.
```
Comment (joelclimbsthings) [inline docs/features/remote-payments.md:9]: Note that I believe this needs to be set pre-build? Either that or you need to make the change in `/includes/feature-config.php`.

File: docs/features/remote-payments.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+Name | Return | Default | Description
+--- | --- | --- | ---
+`needs_setup()` | boolean | `false` | Used to determine if the gateway still requires setup in order to be used.
+`get_required_settings_keys()` | array | `[]` | An array of keys for fields required to properly configure the gateway.  The keys must match those of already registered form fields in the payment gateway.
+`get_oauth_connection_url()` | string | `null` | The oAuth connection URL to be used to quickly connect a payment gateway provider.  If provided, this will be used in place of required setting fields.
+`get_post_install_script()` | string | `null` | A script URL to be dynamically loaded into a page after a gateway has been installed.  This is primarily used for `SlotFill`, but can allow any script to be added to assist in payment gateway setup.
+
+## SlotFill
+
+Payment gateway tasks can be SlotFilled to provide custom experiences.  This is useful if a gateway cannot follow the generic payment steps to be fully set up.  The entire payment gateway card can be SlotFilled using `WooRemotePayment` or simply SlotFill `WooRemotePaymentSettings` to leave the default installation and stepper in place.
```
Comment (joelclimbsthings) [inline docs/features/remote-payments.md:56]: This all looks perfect, @joshuatf . I've also got these components documented with the `@woocommerce/components` package, which I'd imagine will be kept more up-to-date. I wonder if we should link to that here?

File: docs/features/remote-payments.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+Additional information is added to the existing payment gateway in the WooCommerce REST API response.  The following public methods can be added to the payment gateway class to pass information to the recommended payment gateways task:
+
+Name | Return | Default | Description
+--- | --- | --- | ---
+`needs_setup()` | boolean | `false` | Used to determine if the gateway still requires setup in order to be used.
+`get_required_settings_keys()` | array | `[]` | An array of keys for fields required to properly configure the gateway.  The keys must match those of already registered form fields in the payment gateway.
+`get_oauth_connection_url()` | string | `null` | The oAuth connection URL to be used to quickly connect a payment gateway provider.  If provided, this will be used in place of required setting fields.
+`get_post_install_script()` | string | `null` | A script URL to be dynamically loaded into a page after a gateway has been installed.  This is primarily used for `SlotFill`, but can allow any script to be added to assist in payment gateway setup.
```
Comment (joelclimbsthings) [inline docs/features/remote-payments.md:52]: Perhaps mention here that it needs to be the absolute url.

**Review by joelclimbsthings — DISMISSED** [review]
Overall this looks great @joshuatf , thanks for putting this together. Just left a couple minor comments 👌🏻

File: docs/features/remote-payments.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
+# WooCommerce Remote Payment Gateway Recommendations
+
+This feature allows a remote data source to specify the currently recommended payment gateways.  The feature polls the remote JSON file to retrieve a list of recommended and visible gateways based on rulesets to be shown to the user.
+
+Plugins from this source will then walk through an installer step, followed by a connection step with the minimum required fields for setup defined by the downloaded plugin.
```
Comment (psealock) [inline docs/features/remote-payments.md:5]:
```suggestion
After merchants click on a recommendation, plugins from this source will then walk through an installer step, followed by a connection step with the minimum required fields for setup defined by the downloaded plugin.
```

File: docs/features/remote-payments.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+	"is_visible": [
+		<Rule>,
+		...
+	]
+  }
+  ...
+]
+```
+
+The remote specs use the [rule processor](https://github.com/woocommerce/woocommerce-admin/blob/main/src/RemoteInboxNotifications/README.md#rule) to determine if a gateway should be shown using the `is_visible`) property.
```
Comment (psealock) [inline docs/features/remote-payments.md:39]:
```suggestion
The remote specs use the [rule processor](https://github.com/woocommerce/woocommerce-admin/blob/main/src/RemoteInboxNotifications/README.md#rule) to determine if a gateway should be shown using the `is_visible` property.
```

File: docs/features/remote-payments.md (hunk 1 above)
Comment (psealock) [reply docs/features/remote-payments.md:9]: Oof, I just lost 30 minutes of my day debugging because I changed the config _after_ running npm start 😢

**Review by psealock — APPROVED** [review]
Looking good @joshuatf, very useful. I'm pre-approving but I have a couple small things. I also think @joelclimbsthings's suggestion about linking to the component makes more sense than having the markup here as well.

File: docs/features/remote-payments.md (hunk 1 above)
Comment (psealock) [reply docs/features/remote-payments.md:9]: Maybe we can add those config files to what webpack watches for changes

Comment (joelclimbsthings) [reply docs/features/remote-payments.md:9]: I actually often just alter `features.php`, when in doubt. Is that bad? 😨

**Comment by joshuatf** [comment]
Thanks for all the great feedback and suggestions, everyone!  I've added all the suggestions to the docs.

**Pull request merged**